Tasks/CopyFiles/task.json

```diff
@@ -0,0 +1,87 @@
+﻿{
+  "id": "5BFB729A-A7C8-4A78-A7C3-8D717BB7C13C",
+  "name": "CopyFiles",
+  "friendlyName": "Copy Files",
+  "description": "Copy files from source folder to target folder using minimatch patterns",
+  "helpMarkDown": "[More Information](http://go.microsoft.com/fwlink/?LinkID=)",
+  "category": "Utility",
+  "visibility": [
+    "Build"
+  ],
+  "author": "Microsoft Corporation",
+  "version": {
+    "Major": 1,
+    "Minor": 0,
+    "Patch": 0
+  },
+  "demands": [
+  ],
+  "minimumAgentVersion": "1.91.0",
+  "groups": [
+        {
+            "name": "advanced",
+            "displayName": "Advanced",
+            "isExpanded": false
+        }
+    ],
+  "inputs": [
+    {
+      "name": "SourceFolder",
+      "type": "string",
+      "label": "Source Folder",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "Source folder to apply copy patterns to"
+    },  
+    {
+      "name": "Contents",
+      "type": "multiLine",
+      "label": "Contents",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "File paths to include as part of the copy. Supports multiple lines or minimatch patterns. [More Information](http://go.microsoft.com/fwlink/?LinkID=)"
+    },
+    {
+      "name": "TargetFolder",
+      "type": "string",
+      "label": "Target Folder",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "Target folder files will copy to"
+    },
+    {
+      "name": "CleanTargetFolder",
+      "type": "boolean",
+      "label": "Clean Target Folder",
+      "defaultValue": "false",
+      "required": false,
+      "helpMarkDown": "Delete all existing files in target folder before copy.",
+      "groupName":"advanced"
+    },
+    {
+      "name": "OverWrite",
+      "type": "boolean",
+      "label": "Over Write",
+      "defaultValue": "false",
+      "required": false,
+      "helpMarkDown": "Relace existing file in target folder",
+      "groupName":"advanced"
+    },
+    {
+      "name": "UseCommonRoot",
+      "type": "boolean",
+      "label": "Use Common Root",
+      "defaultValue": "false",
```

Comment:
I think that makes the most sense.

From: Bryan MacFarlane [mailto:notifications@github.com]
Sent: Tuesday, November 3, 2015 9:08 AM
To: Microsoft/vso-agent-tasks <vso-agent-tasks@noreply.github.com>
Cc: Chris Patterson <Chris.Patterson@microsoft.com>
Subject: Re: [vso-agent-tasks] Users/tihuang/newartifact (#698)


In Tasks/CopyFiles/task.json<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fMicrosoft%2fvso-agent-tasks%2fpull%2f698%23discussion_r43775933&data=01%7c01%7cChris.Patterson%40microsoft.com%7ccedd2d932b8440f7104e08d2e4714602%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=jKbsfCCX9g%2b4FA5TXWDMninUtPeg5Q66fCnffDuLxTo%3d>:

> +      "groupName":"advanced"

> +    },

> +    {

> +      "name": "OverWrite",

> +      "type": "boolean",

> +      "label": "Over Write",

> +      "defaultValue": "false",

> +      "required": false,

> +      "helpMarkDown": "Relace existing file in target folder",

> +      "groupName":"advanced"

> +    },

> +    {

> +      "name": "UseCommonRoot",

> +      "type": "boolean",

> +      "label": "Use Common Root",

> +      "defaultValue": "false",

Actually, I think since this is a new task, we should start minimal and add as we're sure and it's solid. With that line of thinking, we should remove the common root. Leave the implementation in the code but hard code to false and for now don't expose.

Once this is out there, we can't take it back.

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fMicrosoft%2fvso-agent-tasks%2fpull%2f698%2ffiles%23r43775933&data=01%7c01%7cChris.Patterson%40microsoft.com%7ccedd2d932b8440f7104e08d2e4714602%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=imfzsQcE0JdMZGNArt4gEen%2bAI0wSnVJPSPBWmvUfU4%3d>.